lib/workers/repository/process/lookup/index.ts

```diff
@@ -384,6 +384,24 @@ export async function lookupUpdates(
           // TODO #7154
           update.newDigest =
             update.newDigest ?? (await getDigest(config, update.newValue))!;
+
+          // If the digest could not be determined, report this as otherwise the
+          // update will be omitted later on without notice.
+          if (update.newDigest === null) {
+            logger.debug(
+              { depName, ...update },
+              'Could not determine new digest for update.'
+            );
+
+            // Only report a warning if there is a current digest.
+            // Context: https://github.com/renovatebot/renovate/pull/20175#discussion_r1102615059.
+            if (currentDigest) {
+              res.warnings.push({
+                message: `Could not determine new digest for update (${update.newValue}).`,
```

Comment:
Placing `newValue` in the `message` will mean that if there are 50 deps with this problem, you'll get 50 unique warnings in the Dashboard etc. We need to confirm if that's what will happen and if it's what we want. Do you have a reproduction repo for this?

Reply:
Are warnings deduped? We could have a warning per datasource I assume, but not sure what's the desired behavior here. If you have that many updates for dependencies that cannot be updated, then likely you need to act on them anyway.

The issue can be reproduced in https://github.com/angular/angular/blob/main/WORKSPACE, but it's just one dependency in the `WORKSPACE` `build_bazel_rules_nodejs`

Reply:
Warnings with the same message are deduped, yes

Reply:
I've switched it to include the datasource.